### ISSUE TITLE
pe:layout: updates don't work properly

### ISSUE THREAD

**Issue by yennor** [primefaces-extensions/primefaces-extensions.github.com#764]
I'm implementing an SPA (single page application) using the pe:layout component.
I'm using Primefaces 8 and Primefaces extensions 8
So the content of the layoutPanes need to be updated using ajax requests while the application runs.
So the whole construct looks more or less like that:
```
<h:form id="mainForm">
  <pe:layout fullPage="true" id="layout">
    <pe:layoutPane position="north">
    </pe:layoutPane>
    <pe:layoutPane position="west">
    </pe:layoutPane>
    <pe:layoutPane position="center">
      <ui:include src="#{indexView.centerView}" />
    </pe:layoutPane>
</h:form>
```
So now I want to update the center pane. I do that by updating the payoutPane element:
```
<p:menuitem value="User"
  actionListener="#{indexView.showSettingsUser}"
update=":mainForm:center" />
```
When I do that, the center pane gets updated. But it gets moved to the top of the screen. Beeing displayed over the north pane.

So I try to update the whole layout component:
<p:menuitem value="User"
  actionListener="#{indexView.showSettingsUser}"
update=":mainForm:layout" />
When I do that, absolutely nothing happns.
The only way that seems to work is updating everything at once:
<p:menuitem value="User"
  actionListener="#{indexView.showSettingsUser}"
update=":mainForm" />

**Comment by melloware** [comment]
@yennor Do you mind making me a small reproducer so I can see the problem and debug it using the Primefaces Test project? 
https://github.com/primefaces/primefaces-test

You can execute the sample with mvn jetty:run command and hit http://localhost:8080/primefaces-test to run the page.

**Comment by Rapster** [comment]
I think it's a problem I faced several times using this component. As a workaround, I used CSS. I thought `p:layout` and` pe:layout` components were deprecated?

**Comment by melloware** [comment]
Can you try this..

```xml
<pe:layoutPane position="center">
      <p:outputPanel id="pnlCenter"
      <ui:include src="#{indexView.centerView}" />
      </p:outputPanel>
    </pe:layoutPane>
```

And then doing `update="@root:@id(pnlCenter)"` or `update=":mainForm:layout:pnlCenter"`

So its updating an outputPanel inside the pane and NOT the pane itself?

**Comment by melloware** [comment]
@Rapster PF has deprecated p:layout but PFE will still support pe:layout

**Comment by yennor** [comment]
you're way too fast in replying ;-).

I've just noticed. When I update everything (the whole form) and there is a west layoutPanel. Then It will be at once rendered between north and center panel, instead of being to the left of the center panel.

I've tried out the workaround, using the outputPanel. Like that it works like a charm. :-)
I should manage to provide a test case with the test case.

**Comment by yennor** [comment]
I've did a really quick and dirty example:
https://github.com/yennor/primefaces-test

just. The west position is shown from the beginning between north and center. For whatever reason? It doesn't do that in my environment..
And update on layoutpanel doesn't work (it does in my environment, but moves the central pane).
Only update on form works... anyway.. i guess even like that it doesn't behave the way it should?

**Comment by melloware** [comment]
No probably not but i am not sure whether the panes were meant to be updated or using the OutputPanel trick the content inside the panes to be updated.

**Comment by yennor** [comment]
If they are meant to be used with the outputpanel trick, then that really should be documented at an obvious place.

But as you can see in my example, even before updating anything, the west panel is displayed between the north and the south panel. Which in any case shouldn't happen.

**Comment by melloware** [comment]
Trying to run your reproducer just creates a stack trace for me..

```
javax.servlet.ServletException: javax.servlet.ServletException: 
/test.xhtml @24,50 <ui:include src="#{testView.testString}"> Invalid path : center.xhtml
```

**Comment by yennor** [comment]
sorry. my fault.. forgot to add the new files. they are commited now.

**Comment by melloware** [comment]
Yeah you are right something is funky.  I am not having this issue on the Showcase which uses the layout but this was built all long before I got here.

**Comment by yennor** [comment]
As I said in my project I only get that issue with west being between north and center after updating the whole form. On first render it is all right. So there must be some special conditions triggering that problem.
Propably the same with my original problem, of the center pane beeing displayed on top of the screen after an update.
